packages/airnode-node/src/evm/fulfillments/withdrawals.ts

```diff
@@ -59,13 +59,13 @@ export const submitWithdrawal: SubmitRequest<Withdrawal> = async (airnodeRrp, re
     `Withdrawal gas limit estimated at ${paddedGasLimit.toString()} for Request:${request.id}`
   );
 
-  // We set aside some ETH to pay for the gas of the following transaction,
-  // send all the rest along with the transaction. The contract will direct
-  // these funds back to the sponsor wallet.
+  // We set aside some ETH to pay for the gas of the following transaction and
+  // return the rest to the sponsor wallet less a remainder, if specified.
+  const remainder = options.withdrawalRemainder ? ethers.BigNumber.from(options.withdrawalRemainder) : 0;
   const txCost = paddedGasLimit.mul(
     options.gasTarget.gasPrice ? options.gasTarget.gasPrice : options.gasTarget.maxFeePerGas!
   );
-  const fundsToSend = currentBalance.sub(txCost);
+  const fundsToSend = currentBalance.sub(txCost).sub(remainder);
```

Comment:
What happens when this becomes negative?

Reply:
That is handled just a few lines below 😄 

https://github.com/api3dao/airnode/pull/1131/files#diff-5ad6e79a7234dfa1621a5894aab96dca60eaa1ee9cc9e3c86869aab83c66c999R71-R76

Reply:
Sorry, I only looked at the diffed changes.